howl/client/howl_client.py

```diff
@@ -0,0 +1,102 @@
+from howl.context import InferenceContext
+import logging
+import time
+from typing import Callable
+
+import numpy as np
+import pyaudio
+import torch
+from howl.model.inference import InferenceEngine
+
+
+class HowlClient:
+    def __init__(self,
+                 engine: InferenceEngine = None,
+                 context: InferenceContext = None,
+                 device: int = -1,
+                 chunk_size: int = 500):
+        self.listeners = []
+        self.chunk_size = chunk_size
+        self.device = self._get_device(device)
+
+        self.engine: InferenceEngine = engine
+        self.ctx: InferenceContext = context
+        self._audio = pyaudio.PyAudio()
+        self._audio_buf = []
+        self.last_data = np.zeros(self.chunk_size)
+
+    @staticmethod
+    def list_pretrained(self):
+        """Show a list of available pretrained models"""
+        print(torch.hub.list('castorini/howl:howl-pip'))
+    
+    def _get_device(device: int):
+        if device == -1:
+            return torch.device('cpu')
+        else:
+            return torch.device('cuda:{}'.format(device))
+
+    def _on_audio(self, in_data):
+        data_ok = (in_data, pyaudio.paContinue)
+        self.last_data = in_data
+        self._audio_buf.append(in_data)
+        if len(self._audio_buf) != 16:
```

Comment:
const as well

Reply:
Any naming suggestion for this too? Also just took it from demo.py, should I maybe pull both out and reuse it in both files?